pkg/services/alerting/models.go

```diff
@@ -21,6 +27,19 @@ type EvalMatch struct {
 	Tags   map[string]string `json:"tags"`
 }
 
+// TimeSeries return a string represent a time series.
+// It concatenates metric name and tags, EX: cpu{region=us, host=01}.
+func (e *EvalMatch) TimeSeries() string {
```

Comment:
GetFullSeriesName?

Reply:
This function is not needed when using the Prometheus datasource.
The Prometheus datasource will format the `EvalMatch.Name` to include the labels https://github.com/grafana/grafana/blob/master/pkg/tsdb/prometheus/prometheus.go#L131

It will, however, format the seriename if you configured the legend formater. If that's a problem I suggest introducing a new Field in `EvalMatch` called RawName.

For other datasources I dont think Prometheus formatting makes much sense. So please update the PR to use `EvalMatch.Name`. The rest looks good so once this is fixed this is ready to be merged.

Reply:
Thanks!

Use `EvalMatch.Metric` now.